cargo-dist/src/init.rs

```diff
@@ -1430,3 +1502,26 @@ where
         table.remove(key);
     }
 }
+
+/// Similar to [`apply_optional_value`][] but specialized to `MacPkgConfig`, since we're not able to work with structs dynamically
+fn apply_optional_mac_pkg(
+    table: &mut toml_edit::Table,
+    key: &str,
+    desc: &str,
+    val: Option<&MacPkgConfig>,
+) {
+    if let Some(mac_pkg_config) = val {
+        let new_item = &mut table[key];
+        let mut new_table = toml_edit::table();
+        if let Some(new_table) = new_table.as_table_mut() {
+            new_table.insert("identifier", toml_edit::value(&mac_pkg_config.identifier));
+            if let Some(location) = &mac_pkg_config.install_location {
+                new_table.insert("install-location", toml_edit::value(location));
+            }
+            new_table.decor_mut().set_prefix(desc);
+        }
```

Comment:
Is there a reason these can't call into apply_optional and friends?

Reply:
Huh, right, I guess I didn't really consider that this is *also* a table we could just `apply_optional_*` into.

Reply:
Would you like me to break this init commit out into a separate PR? Then we can land the rest with the init stuff missing.